### ISSUE TITLE
does not work out of the box

### ISSUE THREAD

**Issue by thayneq** [evertramos/docker-compose-letsencrypt-nginx-proxy-companion#107]
When I clone your git and follow your instructions, I am getting a lot of errors, with just the default configuration. The only modification I have done is to adjust the NGINX location in the .env to "./" because I can't figure out where you are trying to have this point to. Maybe this is the problem? I don't have an NGINX installed because it wasn't a prerequisite of this docker.

Here is what I'm seeing:
```
# ./start.sh 
Error response from daemon: network with name webproxy already exists
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16288  100 16288    0     0   171k      0 --:--:-- --:--:-- --:--:--  172k
Pulling nginx-web (nginx:latest)...
latest: Pulling from library/nginx
Digest: sha256:31b8e90a349d1fce7621f5a5a08e4fc519b634f7d3feb09d53fac9b12aa4d991
Status: Image is up to date for nginx:latest
Pulling nginx-gen (jwilder/docker-gen:latest)...
latest: Pulling from jwilder/docker-gen
Digest: sha256:f86c5374c3384b7de37c221e7fa450df357d2bf473db3eedd0375cf39288d9cc
Status: Image is up to date for jwilder/docker-gen:latest
Pulling nginx-letsencrypt (jrcs/letsencrypt-nginx-proxy-companion:latest)...
latest: Pulling from jrcs/letsencrypt-nginx-proxy-companion
Digest: sha256:3181bf5c683c647c4d53526706ca5bce56b37ff7d8c610bd70e194026d174a9a
Status: Image is up to date for jrcs/letsencrypt-nginx-proxy-companion:latest
Starting nginx-web ... 
Starting nginx-gen ... 
Starting nginx-web
Starting nginx-gen
Starting nginx-letsencrypt ... 
Starting nginx-gen ... error

Starting nginx-letsencrypt ... error

Starting nginx-web ... error

ERROR: for nginx-web  Cannot start service nginx-web: Bind mount failed: '/volume1/docker/docker-compose-letsencrypt-nginx-proxy-companion/html' does not exists

ERROR: for nginx-web  Cannot start service nginx-web: Bind mount failed: '/volume1/docker/docker-compose-letsencrypt-nginx-proxy-companion/html' does not exists

ERROR: for nginx-gen  Cannot start service nginx-gen: Bind mount failed: '/volume1/docker/docker-compose-letsencrypt-nginx-proxy-companion/certs' does not exists

ERROR: for nginx-letsencrypt  Cannot start service nginx-letsencrypt: Bind mount failed: '/volume1/docker/docker-compose-letsencrypt-nginx-proxy-companion/certs' does not exists
ERROR: Encountered errors while bringing up the project.
```

**Comment by evertramos** [comment]
Hello @thayneq,

Thanks for your message.

Could you paste your `.env` file?

**Comment by thayneq** [comment]
it's exactly the same as the default, just with:

```
NGINX_FILES_PATH=./
```

**Comment by evertramos** [comment]
@thayneq 

I asked if you could post your .env not this line only...

but ok... try to set some other path to this variable such as:

```bash
NGINX_FILES_PATH=/home/user/webproxy
```

and try again...

The error as your logs says it binding your volume...

```bash
'/volume1/docker/docker-compose-letsencrypt-nginx-proxy-companion/html'
```

**Comment by thayneq** [comment]
more errors.

`# ./start.sh 
Error response from daemon: network with name webproxy already exists
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16288  100 16288    0     0   174k      0 --:--:-- --:--:-- --:--:--  176k
Pulling nginx-web (nginx:latest)...
latest: Pulling from library/nginx
Digest: sha256:31b8e90a349d1fce7621f5a5a08e4fc519b634f7d3feb09d53fac9b12aa4d991
Status: Image is up to date for nginx:latest
Pulling nginx-gen (jwilder/docker-gen:latest)...
latest: Pulling from jwilder/docker-gen
Digest: sha256:f86c5374c3384b7de37c221e7fa450df357d2bf473db3eedd0375cf39288d9cc
Status: Image is up to date for jwilder/docker-gen:latest
Pulling nginx-letsencrypt (jrcs/letsencrypt-nginx-proxy-companion:latest)...
latest: Pulling from jrcs/letsencrypt-nginx-proxy-companion
Digest: sha256:3181bf5c683c647c4d53526706ca5bce56b37ff7d8c610bd70e194026d174a9a
Status: Image is up to date for jrcs/letsencrypt-nginx-proxy-companion:latest
Removing nginx-web
Removing nginx-gen
Removing nginx-letsencrypt
Recreating 189c6b8d89ab_nginx-gen ... 
Recreating 1d48c84469e2_nginx-web ... 
Recreating 1d48c84469e2_nginx-web
Recreating 189c6b8d89ab_nginx-gen
Recreating 473a18abd193_nginx-letsencrypt ... 
Recreating 1d48c84469e2_nginx-web ... error

ERROR: for 1d48c84469e2_nginx-web  Cannot start service nginx-web: driver failed programming external connectivity on endpoint nginx-web (c3a17ec46b9478af833c21a06904ec1805f8e674be993493b5d8bc63309cRecreating 189c6b8d89ab_nginx-gen ... error

Recreating 473a18abd193_nginx-letsencrypt ... error

ERROR: for 473a18abd193_nginx-letsencrypt  Cannot start service nginx-letsencrypt: Bind mount failed: '/volume1/docker/blah/conf.d' does not exists

ERROR: for nginx-web  Cannot start service nginx-web: driver failed programming external connectivity on endpoint nginx-web (c3a17ec46b9478af833c21a06904ec1805f8e674be993493b5d8bc63309c7ac9): Bind for 0.0.0.0:443 failed: port is already allocated

ERROR: for nginx-gen  Cannot start service nginx-gen: Bind mount failed: '/volume1/docker/blah/html' does not exists

ERROR: for nginx-letsencrypt  Cannot start service nginx-letsencrypt: Bind mount failed: '/volume1/docker/blah/conf.d' does not exists
ERROR: Encountered errors while bringing up the project.`

**Comment by thayneq** [comment]
here is my .env as you requested:
`# cat .env
#
# docker-compose-letsencrypt-nginx-proxy-companion
#
# A Web Proxy using docker with NGINX and Let's Encrypt
# Using the great community docker-gen, nginx-proxy and docker-letsencrypt-nginx-proxy-companion
#
# This is the .env file to set up your webproxy enviornment

#
# Your local containers NAME
#
NGINX_WEB=nginx-web
DOCKER_GEN=nginx-gen
LETS_ENCRYPT=nginx-letsencrypt

#
# Set the IP address of the external access Interface
#
IP=0.0.0.0

#
# Default Network
#
NETWORK=webproxy

# If you want to customize the created network, use the following variable
#NETWORK_OPTIONS="--opt encrypted=true"

#
# Service Network (Optional)
#
# In case you decide to add a new network to your services containers you can set this
# network as a SERVICE_NETWORK
#
# [WARNING] This setting was built to use our `start.sh` script or in that special case
#           you could use the docker-composer with our multiple network option, as of:
#           `docker-compose -f docker-compose-multiple-networks.yml up -d`
#
#SERVICE_NETWORK=webservices

# If you want to customize the created network, use the following variable
#SERVICE_NETWORK_OPTIONS="--opt encrypted=true"

#
# NGINX file path
#
NGINX_FILES_PATH=/volume1/docker/blah

#
# NGINX use special conf files
#
# In case you want to add some special configuration to your NGINX Web Proxy you could
# add your files to ./conf.d/ folder as of sample file 'uploadsize.conf'
#
# [WARNING] This setting was built to use our `start.sh`.
#
# [WARNING] Once you set this options to true all your files will be copied to data
#           folder (./data/conf.d). If you decide to remove this special configuration
#           you must delete your files from data folder ./data/conf.d.
#
#USE_NGINX_CONF_FILES=true

#
# Docker Logging Config
#
# This section offers two options max-size and max-file, which follow the docker documentation
# as follow:
#
# logging:
#      driver: "json-file"
#      options:
#        max-size: "200k"
#        max-file: "10"
#
#NGINX_WEB_LOG_DRIVER=json-file
#NGINX_WEB_LOG_MAX_SIZE=4m
#NGINX_WEB_LOG_MAX_FILE=10

#NGINX_GEN_LOG_DRIVER=json-file
#NGINX_GEN_LOG_MAX_SIZE=2m
#NGINX_GEN_LOG_MAX_FILE=10

#NGINX_LETSENCRYPT_LOG_DRIVER=json-file
#NGINX_LETSENCRYPT_LOG_MAX_SIZE=2m
#NGINX_LETSENCRYPT_LOG_MAX_FILE=10`

**Comment by thayneq** [comment]
here is my .env:

```
# cat .env
#
# docker-compose-letsencrypt-nginx-proxy-companion
#
# A Web Proxy using docker with NGINX and Let's Encrypt
# Using the great community docker-gen, nginx-proxy and docker-letsencrypt-nginx-proxy-companion
#
# This is the .env file to set up your webproxy enviornment

#
# Your local containers NAME
#
NGINX_WEB=nginx-web
DOCKER_GEN=nginx-gen
LETS_ENCRYPT=nginx-letsencrypt

#
# Set the IP address of the external access Interface
#
IP=0.0.0.0

#
# Default Network
#
NETWORK=webproxy

# If you want to customize the created network, use the following variable
#NETWORK_OPTIONS="--opt encrypted=true"

#
# Service Network (Optional)
#
# In case you decide to add a new network to your services containers you can set this
# network as a SERVICE_NETWORK
#
# [WARNING] This setting was built to use our `start.sh` script or in that special case
#           you could use the docker-composer with our multiple network option, as of:
#           `docker-compose -f docker-compose-multiple-networks.yml up -d`
#
#SERVICE_NETWORK=webservices

# If you want to customize the created network, use the following variable
#SERVICE_NETWORK_OPTIONS="--opt encrypted=true"

#
# NGINX file path
#
NGINX_FILES_PATH=/volume1/docker/blah

#
# NGINX use special conf files
#
# In case you want to add some special configuration to your NGINX Web Proxy you could
# add your files to ./conf.d/ folder as of sample file 'uploadsize.conf'
#
# [WARNING] This setting was built to use our `start.sh`.
#
# [WARNING] Once you set this options to true all your files will be copied to data
#           folder (./data/conf.d). If you decide to remove this special configuration
#           you must delete your files from data folder ./data/conf.d.
#
#USE_NGINX_CONF_FILES=true

#
# Docker Logging Config
#
# This section offers two options max-size and max-file, which follow the docker documentation
# as follow:
#
# logging:
#      driver: "json-file"
#      options:
#        max-size: "200k"
#        max-file: "10"
#
#NGINX_WEB_LOG_DRIVER=json-file
#NGINX_WEB_LOG_MAX_SIZE=4m
#NGINX_WEB_LOG_MAX_FILE=10

#NGINX_GEN_LOG_DRIVER=json-file
#NGINX_GEN_LOG_MAX_SIZE=2m
#NGINX_GEN_LOG_MAX_FILE=10

#NGINX_LETSENCRYPT_LOG_DRIVER=json-file
#NGINX_LETSENCRYPT_LOG_MAX_SIZE=2m
#NGINX_LETSENCRYPT_LOG_MAX_FILE=10
```

**Comment by thayneq** [comment]
I guess my question is, are you requiring that I have nginx installed at some other location? If so, why don't you include it in your compose? I do not have the USE_NGINX_CONF_FILES set, so I'm just confused why it's set up this way. Any help is appreciated.

**Comment by evertramos** [comment]
nginx is not a requirement.

About the USE_NGINX_CONF_FILES you should use only if you want some special rules set for the nginx in the webproxy.

**Comment by evertramos** [comment]
What do you have in this /volume1 ?

Please, once again, try to use your home dir where you are sure you have write access to it.

**Comment by thayneq** [comment]
/volume1 is the default volume of the synology NAS that I'm using. I put all my docker projects in /volume1/docker. I am running as root and have write access. I'll try again with my home dir, which is in /volume1/homes/user/

I don't expect that to change the output, but here goes anyway:


```
Recreating 189c6b8d89ab_189c6b8d89ab_189c6b8d89ab_nginx-gen ... error

Recreating 473a18abd193_473a18abd193_473a18abd193_nginx-letsencrypt ... error

Recreating 1d48c84469e2_1d48c84469e2_1d48c84469e2_nginx-web ... error

ERROR: for 1d48c84469e2_1d48c84469e2_1d48c84469e2_nginx-web  Cannot start service nginx-web: Bind mount failed: '/volume1/homes/user/blah/htpasswd' does not exists

ERROR: for nginx-web  Cannot start service nginx-web: Bind mount failed: '/volume1/homes/user/blah/htpasswd' does not exists

ERROR: for nginx-gen  Cannot start service nginx-gen: Bind mount failed: '/volume1/homes/user/blah/certs' does not exists

ERROR: for nginx-letsencrypt  Cannot start service nginx-letsencrypt: Bind mount failed: '/volume1/homes/user/blah/html' does not exists
ERROR: Encountered errors while bringing up the project.
```

**Comment by thayneq** [comment]
1. I do not have "ss" so this command does not work.

2. /volume1 is the root directory that is created when you make a filesystem with synology NAS. I don't want to share the output of this command because it lists a lot of private directories that don't apply to this problem. Is there something specific about /volume1 that you want to know?

3. 
```
# uname -a
Linux hughes 4.4.59+ #23824 SMP PREEMPT Fri Oct 12 18:31:48 CST 2018 x86_64 GNU/Linux synology_denverton_1618+
```

**Comment by thayneq** [comment]
based on the errors, it appears start.sh is expecting some nginx files to appear in the NGINX_FILES_PATH, which is just an empty directory I created, so I'm not surprised this error is happening. Should I install nginx into this directory?

**Comment by evertramos** [comment]
@thayneq 

Once again you don´t need nginx installed in the host... please read some docker docs first.

You are trying to use docker in a synology nas... this is the issue you should look for... 

Try to run a hello world container for the docker documentation and see if it works, the issue is not related to this script. You must be able to create and bind volumes, you might will need to add some required libs to you linux before you get through this.

Hope you can solve it.

**Comment by thayneq** [comment]
I run a lot of dockers on synology, as do many other people. I suggest you look a bit closer before closing this issue. What makes you think this is related to synology? I am able to run the non-compose version of these packages just fine.

**Comment by thayneq** [comment]
I disagree with you calling this invalid.

**Comment by evertramos** [comment]
@thayneq 

I did not mean to offend you or anything... sorry if you misunderstand my words.

**Comment by thayneq** [comment]
ok. I guess I'll go look at https://github.com/buchdag/letsencrypt-nginx-proxy-companion-compose

thanks for trying!

**Comment by nicolailang** [comment]
I too am running on Synology, and I now have this running. It's been a while since I set it up on my production (Ubuntu) server but it seems like Synology does not work same way with rights - and does not created folder automatically. 

Anyway my secret was to run the commands below to create the data directory and set the permissions correctly. 

mkdir -p data/certs
mkdir data/htpasswd
mkdir data/conf.d
mkdir data/vhost.d
mkdir data/html
chgrp -R 101 data
chmod -R g+rwx data

Another issue I was facing was that Synology apparently insists on running a HTTP listener no matter what you enable but I have solved that by mapping external ports 80/443 to Synology 81/444 and made the script configureable. I'll create a PR once I have tested and documented this properly (in a few days I hope).
